packages/core/src/shared/sagemaker/client/sagemaker.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as fs from 'fs'
+import { fromContainerMetadata } from '@aws-sdk/credential-provider-imds'
+import { Service } from 'aws-sdk'
+import globals from '../../extensionGlobals'
+import SageMakerClient, { DescribeDomainRequest, DescribeDomainResponse } from './sagemakerclient'
+import apiConfig = require('./service-2.json')
+import { ServiceOptions } from '../../awsClientBuilder'
+
+export interface SageMakerCookie {
+    authMode?: 'Sso' | 'Iam'
+    expiryTime?: number
+    ssoExpiryTimestamp?: number
+    studioUserProfileName?: string
+    redirectURL?: string
+    AccessToken?: string
+    StudioSessionToken?: string
+}
+
+// Manually expire the token every 5 minutes as token refresh is handled externally by Sagemaker
+const TOKEN_EXPIRY = 5 * 60 * 1000
+const DEFAULT_REGION = 'us-east-1'
+
+export class DefaultSageMakerClient {
+    constructor(private readonly region: string) {}
+
+    private async createSdkClient(): Promise<SageMakerClient> {
+        return (await globals.sdkClientBuilder.createAwsService(
+            Service,
+            {
+                apiConfig: apiConfig,
+                region: this.region,
+                credentials: await fromContainerMetadata()(),
+                endpoint: `https://api.sagemaker.${this.region}.amazonaws.com`,
+            } as ServiceOptions,
+            undefined
+        )) as SageMakerClient
+    }
+
+    public async describeDomain(request: DescribeDomainRequest): Promise<DescribeDomainResponse> {
+        return await (await this.createSdkClient()).describeDomain(request).promise()
+    }
+}
+
+/**
+ * Client that provides utilities to fetch information related to
+ * the current domain & container space.
+ */
+export class SageMakerSpaceClient {
+    private static _instance: SageMakerSpaceClient | undefined
+
+    private cachedCookies: SageMakerCookie | undefined = undefined
+    private cachedProfileArn: string | undefined = undefined
+    private sageMakerClient: DefaultSageMakerClient
+
+    private constructor(sageMakerClient?: DefaultSageMakerClient) {
+        this.sageMakerClient = sageMakerClient ?? new DefaultSageMakerClient(this.getDefaultRegion())
+    }
+
+    public static getQConnectionScopes() {
+        return ['codewhisperer:completions', 'codewhisperer:conversations']
+    }
+
+    public static getInstance(): SageMakerSpaceClient {
+        if (!this._instance) {
+            this._instance = new SageMakerSpaceClient()
+        }
+        return this._instance
+    }
```

Comment:
Follow the same format that we have in other singletons for consistency https://github.com/aws/aws-toolkit-vscode/blob/e6c3101e8950acea5158407b7099d66a26663a8c/packages/core/src/codewhisperer/util/authUtil.ts#L234-L241

```suggestion
    public static get instance() {
        if (!this.#instance) {
            this.#instance = new SageMakerSpaceClient()
        }
        return this.#instance
    }
```